### ISSUE TITLE
read-only flags are not properly inherited in translations

### ISSUE THREAD

**Issue by surli** [WeblateOrg/weblate#8792]
### Describe the issue

I marked a translation as read-only using a flag in the source translation, and I'm now expecting the other languages to display the same flag. However I don't see the flag in other translations. Also the source translation UI now displays that the read-only flag is inherited which does not really make sense. 
See:
![image](https://user-images.githubusercontent.com/1478232/218413562-e93acdf1-f37f-4fbc-8e60-1d6b4b276ec0.png)

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

1. Go to a component having already existing translations
2. In the source of translation, chose a key and mark it read-only using a flag
3. Go to an existing translation of the same key
4. See that the translation is not marked read-only

### Expected behavior

I'm expected the read-only flags to be automatically inherited from the source translation to the translations.  

### Screenshots

Source translation with weird explanation about inherited flag:
![image](https://user-images.githubusercontent.com/1478232/218414005-19806d0c-62a6-4836-8564-d8100bf02787.png)

Translation of the same key:
![image](https://user-images.githubusercontent.com/1478232/218414246-bc258c75-849b-4392-bd48-c0c86846dd94.png)

### How do you run Weblate?

PyPI module

### Weblate versions

 * Weblate: 4.15.1
 * Django: 3.2.4
 * siphashc: 1.3
 * translate-toolkit: 3.8.3
 * lxml: 4.9.2
 * Pillow: 9.1.0
 * bleach: 3.1.5
 * python-dateutil: 2.8.1
 * social-auth-core: 4.3.0
 * social-auth-app-django: 5.0.0
 * django-crispy-forms: 1.9.2
 * oauthlib: 3.1.0
 * django-compressor: 2.4
 * djangorestframework: 3.11.1
 * django-filter: 2.4.0
 * django-appconf: 1.0.4
 * user-agents: 2.1
 * filelock: 3.9.0
 * rapidfuzz: 2.13.7
 * openpyxl: 3.0.5
 * celery: 5.1.0
 * kombu: 5.1.0
 * translation-finder: 2.15
 * weblate-language-data: 2023.1
 * html2text: 2020.1.16
 * pycairo: 1.19.1
 * pygobject: 3.38.0
 * diff-match-patch: 20200713
 * requests: 2.27.1
 * django-redis: 4.12.1
 * hiredis: 1.0.1
 * sentry_sdk: 1.14.0
 * Cython: 0.29.21
 * misaka: 2.1.1
 * GitPython: 3.1.8
 * borgbackup: 1.1.13
 * pyparsing: 3.0.8
 * pyahocorasick: 1.4.2
 * python-redis-lock: 4.0.0
 * charset-normalizer: 2.0.12
 * Python: 3.7.3
 * Git: 2.20.1
 * psycopg2-binary: 2.8.6
 * phply: 1.2.6
 * ruamel.yaml: 0.16.12
 * tesserocr: 2.5.2
 * boto3: 1.22.10
 * zeep: 4.1.0
 * aeidon: 1.11
 * iniparse: 0.5
 * Mercurial: 6.1.2
 * git-review: 2.3.1
 * Redis server: 5.0.14
 * PostgreSQL server: 11.18
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: localhost
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://localhost:6379, redis://localhost:6379, regular
 * Platform: Linux 4.19.0-20-amd64 (x86_64)

**Comment by nijel** [comment]
The source language is read-only in some cases (bilingual format, or disabled editing of it), that's why you see the inherited flag there. There is no explicit flag that should be passed to the translations, though.

**Comment by surli** [comment]
> The source language is read-only in some cases (bilingual format, or disabled editing of it)

oooh ok I got it there's a computed read-only flag provided by the fact I disabled edition on my source language, and on top of that I can add an explicit "read-only" flag that will be inherited in the translations.
Apparently I checked again and I did added the read-only explicit flag: I suspect that I  got some kind of cache issue when I created this ticket. And this explicit flag is apparently properly inherited. So I'm going to close this ticket. Thanks!
